### DISCUSSION TITLE
Panic: runtime error: invalid memory address or nil pointer dereference

### DISCUSSION THREAD

**Discussion by Usstzst** [terraform-linters/tflint#1596] [Q&A] [answered]
# Introduction

When I use tflint to check all terraform files in the current directory，I run the following command:
`find .  -type d | xargs -I {} tflint {} --color`

tflint is working well, but I get part of outputs like this：

> Panic: runtime error: invalid memory address or nil pointer dereference
 -> 0: main.main.func1: /main.go(21)
 -> 1: runtime.gopanic: /panic.go(884)
 -> 2: runtime.panicmem: /panic.go(260)
 -> 3: runtime.sigpanic: /signal_unix.go(835)
 -> 4: github.com/terraform-linters/tflint/terraform.(*Parser).LoadConfigDir: /parser.go(71)
 -> 5: github.com/terraform-linters/tflint/tflint.(*Loader).LoadConfig: /loader.go(65)
 -> 6: github.com/terraform-linters/tflint/cmd.(*CLI).setupRunners: /inspect.go(117)
 -> 7: github.com/terraform-linters/tflint/cmd.(*CLI).inspect: /inspect.go(41)
 -> 8: github.com/terraform-linters/tflint/cmd.(*CLI).Run: /cli.go(98)
 -> 9: main.main: /main.go(34)
 -> 10: runtime.main: /proc.go(250)
 -> 11: runtime.goexit: /asm_amd64.s(1594)

> TFLint crashed... :(
Please attach an output log, describe the situation and version that occurred and post an issue to https://github.com/terraform-linters/tflint/issues
Panic: runtime error: invalid memory address or nil pointer dereference
 -> 0: main.main.func1: /main.go(21)
 -> 1: runtime.gopanic: /panic.go(884)
 -> 2: runtime.panicmem: /panic.go(260)
 -> 3: runtime.sigpanic: /signal_unix.go(835)
 -> 4: github.com/terraform-linters/tflint/terraform.(*Parser).LoadConfigDir: /parser.go(71)
 -> 5: github.com/terraform-linters/tflint/tflint.(*Loader).LoadConfig: /loader.go(65)
 -> 6: github.com/terraform-linters/tflint/cmd.(*CLI).setupRunners: /inspect.go(117)
 -> 7: github.com/terraform-linters/tflint/cmd.(*CLI).inspect: /inspect.go(41)
 -> 8: github.com/terraform-linters/tflint/cmd.(*CLI).Run: /cli.go(98)
 -> 9: main.main: /main.go(34)
 -> 10: runtime.main: /proc.go(250)
 -> 11: runtime.goexit: /asm_amd64.s(1594)

I want to know why and how to resolve this.

# Additional Text 
$ tflint -v
TFLint version 0.41.0
ruleset.terraform (0.1.1-bundled)

$ terraform -v
Terraform v1.3.2
on linux_386

**Comment by wata727** [comment, marked as answer]
Thank you for your report. This seems to be a panic that occurs in limited circumstances. Fixed by https://github.com/terraform-linters/tflint/pull/1598.

Although it is the cause, I think that there is a file that exists but cannot be read.

**Reply by Constantin07** [reply]
I'm also facing this issue https://github.com/terraform-linters/tflint/issues/1646 with `0.44.1`
